vector_quantize_pytorch/vector_quantize_pytorch.py

```diff
@@ -130,19 +189,16 @@ def init_embed_(self, data):
         if self.initted:
             return
 
-        embed, cluster_size = kmeans(data, self.codebook_size, self.kmeans_iters)
+        embed, cluster_size = kmeans(data, self.codebook_size, self.kmeans_iters,
+                                     sample_fn = self.sample_fn, all_reduce_fn = self.all_reduce_fn)
+
         self.embed.data.copy_(embed)
         self.embed_avg.data.copy_(embed.clone())
         self.cluster_size.data.copy_(cluster_size)
         self.initted.data.copy_(torch.Tensor([True]))
 
     def replace(self, samples, mask):
-        modified_codebook = torch.where(
-            mask[..., None],
-            sample_vectors(samples, self.codebook_size),
-            self.embed
-        )
-        self.embed.data.copy_(modified_codebook)
+        self.embed.data[mask] = self.sample_fn(samples, mask.sum().item())
```

Comment:
Isn't `torch.count_nonzero(mask)` supposed to be faster than `mask.sum().item()` here? @lucidrains @falkaer